src/fps/proposal/MultisigProposal.sol

```diff
@@ -0,0 +1,854 @@
+        require(
+            block.chainid == getChain("mainnet").chainId || block.chainid == getChain("sepolia").chainId,
+            string.concat("Unsupported network: ", vm.toString(block.chainid))
+        );
+
+        string memory taskConfigFileContents = vm.readFile(taskConfigFilePath);
+
+        bytes memory fileContents = vm.parseToml(taskConfigFileContents, ".task");
+        config = abi.decode(fileContents, (TaskConfig));
+
+        name = config.name;
+        description = config.description;
+    }
+
+    /// @notice Sets the L2 networks configuration
+    /// @param networkConfigFilePath Path to the network configuration file
+    /// @param _addresses Address registry contract
+    function setL2NetworksConfig(string memory networkConfigFilePath, Addresses _addresses) public override {
+        addresses = _addresses;
+        string memory networkConfigFileContents = vm.readFile(networkConfigFilePath);
+
+        nonce = abi.decode(vm.parseToml(networkConfigFileContents, ".safeNonce"), (uint256));
+        isNestedSafe = abi.decode(vm.parseToml(networkConfigFileContents, ".isNestedSafe"), (bool));
+
+        /// get chains
+        Addresses.ChainInfo[] memory chains = addresses.getChains();
+        require(chains.length > 0, "MultisigProposal: no chains found");
+
+        /// check that the safe address is the same for all chains and then set safe in storage
+        multisig = addresses.getAddress(config.safeAddressString, chains[0].chainId);
+
+        for (uint256 i = 1; i < chains.length; i++) {
+            require(
+                multisig == addresses.getAddress(config.safeAddressString, chains[i].chainId),
+                string.concat(
+                    "MultisigProposal: safe address mismatch. Caller: ",
+                    vm.getLabel(multisig),
+                    ". Actual address: ",
+                    vm.getLabel(addresses.getAddress(config.safeAddressString, chains[i].chainId))
+                )
+            );
+        }
+
+        /// Fetch starting owners
+        IGnosisSafe safe = IGnosisSafe(multisig);
+        startingOwners = safe.getOwners();
+
+        for (uint256 i = 0; i < config.allowedStorageWriteAccesses.length; i++) {
+            for (uint256 j = 0; j < chains.length; j++) {
+                _allowedStorageAccesses.add(
+                    addresses.getAddress(config.allowedStorageWriteAccesses[i], chains[j].chainId)
+                );
+            }
+        }
+    }
+
+    /// @notice function to be used by forge script.
+    /// @dev use flags to determine which actions to take
+    ///      this function shoudn't be overriden.
+    function _processProposal() internal override {
+        mock();
+        build();
+        simulate();
+        validate();
+        print();
+    }
+
+    /// @notice get the calldata to be executed by safe
+    /// @return data The calldata to be executed
+    function getCalldata() public view override returns (bytes memory data) {
```

Comment:
Seems like this function is only callable when the actions have been set. Otherwise it reverts with `No actions found`. Is it possible to make the function description a little clearer on when this function would be eligible for successful execution.  Seems like `build();` _must_ be called first.